internal/mail/gmail/gmailauth.go

```diff
@@ -0,0 +1,174 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information
+
+package gmail
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/smtp"
+	"net/url"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/zeebo/errs"
+)
+
+// Auth is XOAUTH2 implementation of smtp.Auth interface for gmail
+type Auth struct {
+	UserEmail string
+
+	Storage *TokenStore
+}
+
+// Start returns proto and auth credentials for first auth msg
+func (auth *Auth) Start(server *smtp.ServerInfo) (proto string, toServer []byte, err error) {
+	token, err := auth.Storage.Token()
+	if err != nil {
+		return "", nil, err
+	}
+
+	format := fmt.Sprintf("user=%s\001auth=%s %s\001\001", auth.UserEmail, token.Type, token.AccessToken)
```

Comment:
This escaping looks wrong. Also use hex values rather than octal, they should be a bit clearer to read.

Reply:
I don't quite get why should we escape here. Token and it's type comes from the server it's not like you have to modify them. And if user will input bad data, then parsing will fail and you will receive authentication error. Or did I miss something?

Reply:
fair enough :)